src/sinol_make/commands/export/__init__.py

```diff
@@ -0,0 +1,287 @@
+from sinol_make.interfaces.BaseCommand import BaseCommand
+
+
+class Command(BaseCommand):
+    """
+    Class for "export" command.
+    """
+
+    def get_name(self):
+        return "export"
+
+    def configure_subparser(self, subparser: argparse.ArgumentParser):
+        parser = subparser.add_parser(
+            self.get_name(),
+            help='Create archive for oioioi upload',
+            description='Creates archive ready to upload to oioioi. '
+                        'It is possible to create ocen and dlazaw debian packages for usage in contests '
+                        'with --ocen flag.'
+        )
+
+        parser.add_argument('-o', '--output', type=str, default='export',
+                            help='output directory. Default: export')
+        parser.add_argument('--ocen', action='store_true', default=False,
+                            help='create ocen (or dlazaw in case `dlazaw/` directory exists) debian package '
+                                 'for usage in contests')
+        parsers.add_compilation_arguments(parser)
+
+    def get_generated_tests(self):
+        """
+        Returns list of generated tests.
+        Executes ingen to check what tests are generated.
+        """
+        with tempfile.TemporaryDirectory() as tmpdir:
+            ingen_path = gen_util.get_ingen(self.task_id)
+            if os.path.splitext(ingen_path)[1] == '.sh':
+                ingen_exe = ingen_path
+            else:
+                ingen_exe = gen_util.compile_ingen(ingen_path, self.args, self.args.weak_compilation_flags)
+            if not gen_util.run_ingen(ingen_exe, tmpdir):
+                util.exit_with_error('Failed to run ingen.')
+
+            tests = glob.glob(os.path.join(tmpdir, f'{self.task_id}*.in'))
+            return [package_util.extract_test_id(test) for test in tests]
+
+    def copy_files(self, target_dir: str):
+        """
+        Copies files from current directory to target directory.
+        :param target_dir: Directory to copy files to.
+        """
+        files = ['config.yml', 'makefile.in', 'Makefile.in',
+                 'prog', 'doc', 'public', 'dlazaw']
+        for file in files:
+            file_path = os.path.join(os.getcwd(), file)
+            if os.path.exists(file_path):
+                if os.path.isdir(file_path):
+                    shutil.copytree(file_path, os.path.join(target_dir, file))
+                else:
+                    shutil.copy(file_path, target_dir)
+
+        print('Generating tests...')
+        generated_tests = self.get_generated_tests()
+        print('Copying only handwritten tests...')
+        for ext in ['in', 'out']:
+            os.mkdir(os.path.join(target_dir, ext))
+            for test in glob.glob(os.path.join(os.getcwd(), ext, f'{self.task_id}*.{ext}')):
+                if package_util.extract_test_id(test) not in generated_tests:
+                    shutil.copy(test, os.path.join(target_dir, ext))
```

Comment:
I think that tests `abc0*.in` should always be copied, as often the latex task statement requires them to exist to compile (because e.g. of the sample input/outputs in the statement). And also, handwritten tests without ingen might be a bit of a bad practice, we should discuss whether we want to discourage it or whether it is fine. What do you think?

Reply:
I just saw a OIJ package with all tests generated and no ingen. I don't have great knowledge about creating tasks, but I think there are cases that handwritten tests may be helpful

Reply:
For example sometimes tests can be written during the contest for specific bad solutions

Reply:
Sounds good.